countfiles/utils/file_handlers.py

```diff
@@ -96,3 +99,65 @@ def get_files_without_extension(path: str, recursive=False, include_hidden=True)
         else:
             return [f for f in Path(path).iterdir()
                     if f.is_file() and not is_hidden(f) and not has_extension(f)]
+
+
+def is_hidden_file_or_dir(platform_name: str, path: str) -> bool:
+    """The function determines whether the file or folder is hidden.
+
+    Windows: testing the FILE_ATTRIBUTE_HIDDEN for file or folder
+    (source: https://stackoverflow.com/questions/284115/cross-platform-hidden-file-detection)
+    Linux: testing at least for the dot character in path on Unix-like systems
+    Note: for Linux def is_hidden_file_or_dir('~/path/.to/file.txt') checking for the dot character in path,
+    so if '/.' in path the entire folder is ignored, even if it has visible files.
+    For Windows def is_hidden_file_or_dir('~/path/to/file.txt')
+    checking only file. For checking folder - def is_hidden_file_or_dir('~/path/to_folder')
+    :param platform_name: result of a 'sys.platform' parameter call
+    :param path: ~/path/to/file.txt or ~/path/to_folder
+    :return: True if hidden or False if not
+    """
+    if platform_name.startswith('win'):
+        return bool(os.stat(path).st_file_attributes & stat.FILE_ATTRIBUTE_HIDDEN)
+    elif platform_name.startswith('linux'):
+        return bool('\.' in path or '/.' in path)
+
+
+# TODO: add checking for hidden folder in Windows to skip it.
+def non_recursive_search(location: str, platform_name: str, hidden: bool) -> List[str]:
+    """Non recursive search for files in folder.
+
+    :param location: ~/path/to_folder
+    :param platform_name: result of a 'sys.platform' parameter call
+    :param hidden: False -> exclude hidden, True -> include hidden, counting all files
+    :return: list with filenames
+    """
+    with os.scandir(location) as directory:
+        # if True return all files
+        if hidden:
+            result = [f for f in directory if f.is_file()]
+        else:
+            result = [f for f in directory if f.is_file()
+                      and not is_hidden_file_or_dir(platform_name, os.path.join(location, f))]
+    return result
+
+
+def recursive_search(location: str, platform_name: str, hidden: bool) -> List[str]:
+    """Recursive search for files in folder and subfolders.
+
+    parts[2] - list with filenames, parts[0] - ~/path/to/file
+    :param location: ~/path/to_folder
+    :param platform_name: result of a 'sys.platform' parameter call
+    :param hidden: False -> exclude hidden, True -> include hidden, counting all files
+    :return: list with filenames
+    """
+    if hidden:
+        result = []
+        for parts in os.walk(location):
```

Comment:
Shouldn't we keep, for readability's sake:

```
for root, dirs, files in os.walk()
        result.extend(files)
```

It's, in my opinion, a little more readable because we are name things what they are. But in case this is somewhat slower, we then have a good justification tho change it to the generic `parts`.

Reply:
About speed, I do not know